### PR TITLE
Added promise returns to Plotly.___

### PR DESCRIPTION
**Pull request by mdtusz**
Added promise returns for exposed methods in `plot_api.js`, and they will all be sent along with the gd.

https://github.com/plotly/plotly.js/issues/76

Tests on `npm run test-image` are not passing on my machine, but let's see if they pass on circle!

### CODE REVIEW

File: src/plot_api/plot_api.js [hunk 1]
```diff
@@ -1640,7 +1650,7 @@ Plotly.restyle = function restyle(gd, astr, val, traces) {
     }
     else {
         console.log('restyle fail',astr,val,traces);
-        return;
+        return new Promise.reject();
```
Comment (etpinard) [inline src/plot_api/plot_api.js]: :tada:

**Comment by etpinard** [comment]
Nice. 

Let's get a test case in `test/jasmine/tests/plot_api_test.js` checking that the `gd` is in fact passed into the the `Plotly.plot.then`.

File: test/jasmine/tests/plot_api_test.js [hunk 2]
```diff
@@ -188,65 +181,60 @@ describe('Test graph_obj', function () {
...
+            var expectedData = [
+                {'name': 'a'},
+                {'name': 'b'},
+                {'name': 'c'},
+                {'name': 'd'}
+            ];
 
             Plotly.addTraces(gd, [{'name': 'c'}, {'name': 'd'}]);
-            expect(gd).toEqual(expected);
+            expect(gd.data).toEqual(expectedData);
```
Comment (etpinard) [inline test/jasmine/tests/plot_api_test.js]: FYI @bpostlethwaite no need to check the whole `gd`, just checking its data is more than enough here.

File: test/jasmine/tests/plot_api_test.js [hunk 3]
```diff
@@ -63,11 +63,10 @@ describe('Test graph_obj', function () {
                     {'name': 'd'}
                 ]
             };
-            Plotly.Queue = null;
```
Comment (mdtusz) [inline test/jasmine/tests/plot_api_test.js]: NONONONONONO.

Comment (etpinard) [inline test/jasmine/tests/plot_api_test.js]: Let's never do that again.

File: test/jasmine/tests/plot_api_test.js (hunk 2 above)
Comment (bpostlethwaite) [inline test/jasmine/tests/plot_api_test.js]: nice catch, agreed!

**Comment by etpinard** [comment]
:dancer:

**Pull request merged**